core/ecschema-editing/src/Editing/ChangeInfo/ChangeInfo.ts

```diff
@@ -0,0 +1,273 @@
+import { SchemaItemType } from "@itwin/ecschema-metadata";
+import { SchemaContextEditor } from "../Editor";
+import { ChangeOptionProps, ChangeOptions } from "./ChangeOptions";
+import { RenamePropertyChange } from "./RenamePropertyChange";
+import { SchemaEditType } from "../SchmaEditType";
+
+
+export type SchemaChangeRevertCallback = <T extends ISchemaEditChangeInfo>(changeInfo: T) => Promise<void>;
+export type BeginSchemaEditCallback= <T extends ISchemaEditChangeInfo>(changeInfo: T) => Promise<boolean>;
+
+export interface ISchemaEditChangeProps {
+  readonly changeOptions?: ChangeOptionProps;
+  readonly editType?: string;
+  readonly schemaItemType?: string;
+}
+
+export interface ISchemaEditChangeInfo {
+  readonly contextEditor: SchemaContextEditor;
+  readonly changeOptions: ChangeOptions;
+  readonly editType: SchemaEditType;
+  readonly schemaItemType: SchemaItemType;
+  set sequence(value: number)
+  get sequence(): number;
```

Comment:
if we both have a public getter and setter, we could just defined it as property.
`sequence: number`